app/ui/src/main/java/com/fsck/k9/activity/setup/AccountDiscovery.kt

```diff
@@ -0,0 +1,134 @@
+package com.fsck.k9.activity.setup
+
+import com.fsck.k9.mail.AuthType
+import com.fsck.k9.mail.ConnectionSecurity
+import com.fsck.k9.mail.ServerSettings
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.launch
+import org.jetbrains.anko.coroutines.experimental.bg
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserException
+import org.xmlpull.v1.XmlPullParserFactory
+import timber.log.Timber
+import java.io.IOException
+import java.io.InputStream
+import java.io.InputStreamReader
+import java.net.HttpURLConnection
+import java.net.URL
+import java.net.URLEncoder
+
+class AccountDiscovery {
+
+    class DiscoveredSettings(val incoming: ServerSettings, val outgoing: ServerSettings)
+
+    fun discover(email: String, callback: (settings: DiscoveredSettings?) -> Unit) {
+        launch(UI) {
+            var settings: DiscoveredSettings? = null
+            try {
+                settings = bg {
+                    val url = URL(getAutodiscoveryAddress(email))
```

Comment:
Please use OkHttp as HTTP client.

Reply:
:thinking: I thought it would be as easy as adding new dependency and using the API but after some time adjusting okhttp rules the compilation still fails with `Invoke-customs are only supported starting with android 0 -- min-api 26`. I'll try to get this fixed soon, for now I leave the fetching code as-is.